metric_test.go

```diff
@@ -20,3 +20,30 @@ func BenchmarkSetId(b *testing.B) {
 		metric.SetId()
 	}
 }
+
+func TestTagValidation(t *testing.T) {
+	type testCase struct {
+		tag       []string
+		expecting bool
+	}
+
+	testCases := []testCase{
+		{[]string{"abc=cba"}, true},
+		{[]string{"a="}, false},
+		{[]string{"a!="}, false},
+		{[]string{"=abc"}, false},
+		{[]string{"@#$%!=(*&"}, false},
+		{[]string{"!@#$%=(*&"}, false},
+		{[]string{"@#;$%=(*&"}, false},
+		{[]string{"@#$%=(;*&"}, false},
+		{[]string{"@#$%=(*&"}, true},
+		{[]string{"@#$%=(*&", "abc=!fd", "a===="}, true},
+		{[]string{"@#$%=(*&", "abc=!fd", "a===;="}, false},
```

Comment:
can you add a test case for multiple = e.g. `foo=bar=baz` should be false

Reply:
actually that's legal

Reply:
what's the benefit of allowing this?

Reply:
a value can hold any character besides `;`, if the user has a reason for wanting to use `=` then they're allowed to.  Similarly a key can contain any character besides `=` or `;`. If we want to make that more restrictive we'll want to update graphite-web and carbon to do the same.

https://github.com/graphite-project/graphite-web/blob/master/webapp/graphite/tags/utils.py#L47

Reply:
hmm that's a very wide allowance. at least we should prevent non-printable characters.
(go makes this easy via https://golang.org/pkg/unicode/#IsPrint )
 how about whitespace? will `foo= bar ` result in a value that has a space before and after bar?
there's also so many symbols and special chars.  I generally believe in "if you prevent people from doing dumb things, you also prevent them from doing smart things" which in this setting translates to "allow as much as possible to empower advanced use cases we haven't thought of yet"; but it seems much more common that people will accidentally include the wrong characters such as backspaces, bell, etc, rather than people legitimately trying to use them.

Reply:
independently of what makes sense i think we should use graphite as the reference implementation. so if we want to change the rules we should first change them in graphite, and then everywhere else.

Reply:
@DanCech any thoughts on the above? otherwise this pr looks good to me.